bindings/azure/signalr/signalr.go

```diff
@@ -22,103 +22,168 @@ import (
 	"strings"
 	"time"
 
-	"github.com/golang-jwt/jwt"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	jwt "github.com/golang-jwt/jwt/v4"
 	"github.com/pkg/errors"
 
+	azauth "github.com/dapr/components-contrib/authentication/azure"
 	"github.com/dapr/components-contrib/bindings"
 	"github.com/dapr/kit/logger"
 )
 
 const (
-	errorPrefix         = "azure signalr error:"
-	logPrefix           = "azure signalr:"
+	errorPrefix = "azure signalr error:"
+	logPrefix   = "azure signalr:"
+
+	// Metadata keys.
+	// Azure AD credentials are parsed separately and not listed here.
 	connectionStringKey = "connectionString"
+	accessKeyKey        = "accessKey"
+	endpointKey         = "endpoint"
 	hubKey              = "hub"
-	groupKey            = "group"
-	userKey             = "user"
+
+	// Invoke metadata keys.
+	groupKey = "group"
+	userKey  = "user"
 )
 
-// NewSignalR creates a new pub/sub based on Azure SignalR.
+// Global HTTP client
+var httpClient *http.Client
+
+func init() {
+	httpClient = &http.Client{
+		Timeout: 30 * time.Second,
+	}
+}
+
+// NewSignalR creates a new output binding for Azure SignalR.
 func NewSignalR(logger logger.Logger) *SignalR {
 	return &SignalR{
-		tokens:     make(map[string]signalrCachedToken),
-		httpClient: &http.Client{Timeout: 30 * time.Second},
 		logger:     logger,
+		httpClient: httpClient,
 	}
 }
 
-type signalrCachedToken struct {
-	token      string
-	expiration time.Time
-}
-
 // SignalR is an output binding for Azure SignalR.
 type SignalR struct {
-	endpoint   string
-	accessKey  string
-	version    string
-	hub        string
-	userAgent  string
-	tokens     map[string]signalrCachedToken
```

Comment:
We don't need to cache tokens generated via SHA-256. The way the cache was implemented was not safe for concurrent use, and regardless, signing a JWT requires 1 iteration of SHA-256 which is incredibly fast... so better save the memory :)